src/io/configreader.hpp

```diff
@@ -748,18 +620,17 @@ void ConfigReader<T>::load(std::string &specifier) {
           model->block_index_map.insert(
               {static_cast<std::string>(heart_outlet_junction_name),
                block_count});
-          block_count++;
-          for (auto heart_outlet_block : closed_loop_config["outlet_blocks"]) {
-            connections.push_back(
-                {heart_outlet_junction_name, heart_outlet_block});
+          for (auto outlet_block :
+               closed_loop_config.get_string_array("outlet_blocks")) {
+            connections.push_back({heart_outlet_junction_name, outlet_block});
+            block_count++;
           }
         } else {
           throw std::runtime_error(
               "Error. Only one ClosedLoopHeartAndPulmonary can be included.");
         }
       }
     }
-  } catch (simdjson::simdjson_error) {
   }
 
```

Comment:
Just to make sure errors like the above ones don't happen, can you add in a check to make sure the length of `model->blocks`, `model->block_index_map`, and anywhere else that we are creating a vector related to the number of blocks are all the same size and also equal to `block_count`? Throw a runtime error if there is a mismatch in these vector lengths.

Reply:
Without understanding the details, is that something the tests should have caught? Is `block_count` currently one in all test cases?

Reply:
That's a good idea, I added two sanity checks to the ConfigReader.

Well, the tests we have are not super extensive tbh. They test for analytical correctness, which is excellent, but I think in the long run, we also need additional integration tests, where an actual 0D model is tested and the result is compared for every time step. But I don't think that's within the scope of this PR.